app/jobs/google/distribute_search_job.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Google
+  class DistributeSearchJob < ApplicationJob
+    queue_as :default
+
+    def perform(keywords)
```

Comment:
@malparty Usually we never pass whole objects in active job, the reason being that when job is enqued this part is stored in memory. So we should just pass the id or ids.  Refer to this code as example https://github.com/nimblehq/mia-web/blob/development/app/jobs/payments/campaign_charge_processing_job.rb

Reply:
Yes, `sidekiq` itself recommends passing `id` instead of passing object, the object might gets updated in the database before it process in the job, then the job will process the outdated object. https://github.com/mperham/sidekiq/wiki/Best-Practices#1-make-your-job-parameters-small-and-simple

Reply:
I'm just passing Ids (a hash containing :id, :name & :user_id). Not the full keyword. 
But while working on the notifications, I realized that my tests were not passing a hash (while the real code does). 

So I will have to review this PR -- for that reason I'm setting it back to WIP.

Thanks a lot for your insights @ankitkalia1195  @junan !! :)

Reply:
@malparty Why do we need to pass the hash as well, just list of `keyword_ids` should be enough?

Reply:
Well, if we pass the id only, each search job will need 1 additional SQL query (`select name from keyword where id XXX`). 
While the insset_all fonction enable to natively return more than the `id` column (with the `RETURNING` keyword), I assume that a int + 2x string less than 255char remains quite small. 
My initial arbitration was mainly around: SQL server charge impacts users directly while scheduled tasks charge does not. 

But after reading your article, I understand that even a named property is not recommended as it contains the state of the object (that might change from the time being queued to being processed). 

> [...] Don't pass symbols, named parameters, [...]

I will update accordingly to have only the `keyword_id` triggering the jobs. ;-)

Reply:
@ankitkalia1195  @junan 
Code updated accordingly 👍

Note that I've also added a test in the `CSVUploadForm` to assert the returning keyword_ids array ;-)